### PR TITLE
Don't abort on missing type mappings

### PR DESCRIPTION
**Pull request by nightscape**
Also make ResultSet iteration a little nicer.

### CODE REVIEW

File: src/main/scala/com/geirsson/codegen/Codegen.scala [hunk 1]
```diff
@@ -54,62 +54,73 @@ case class Codegen(options: CodegenOptions, namingStrategy: NamingStrategy) {
...
-        val name = rs.getString(TABLE_NAME)
-        sb += Table(
+    results(rs).flatMap { row =>
+      val name = row.getString(TABLE_NAME)
+      if (!excludedTables.contains(name)) {
+        val columns = getColumns(db, name, foreignKeys)
+        val mappedColumns = columns.filter(_.isRight).map(_.right.get)
+        val unmappedColumns = columns.filter(_.isLeft).map(_.left.get)
+        if (!unmappedColumns.isEmpty)
+          println(s"The following columns from table $name need a mapping: $unmappedColumns")
```
Comment (olafurpg) [inline src/main/scala/com/geirsson/codegen/Codegen.scala]: Maybe turn `println` into a `warn`?

``` scala
def warn(msg: String): Unit = {
  System.err.println(s"[${Console.YELLOW}warn${Console.RESET}] $msg")
}
```

I would like stdout to only be the generated source code for use in pipes, etc.

**Comment by olafurpg** [comment]
Looks great, I agree it's better to not exit on a missing type mapping.

One last thing before we merge, maybe add a new column to the `articles` table in the test? The could could have a type like `circle` just to see that nothing crashes and the generated sources remain the unchanged.

**Comment by olafurpg** [comment]
Actually, maybe it's OK to just fall back to the db column type name as a scala type name instead of aborting or skipping those columns. Replace spaces with underscores, just in case. It might be more convenient to configure type aliases in Scala than to pass in a long `--type-map a,b;c,d;...` argument in the CLI or a big `Map[String, String]` via the library. WDYT?

File: src/main/scala/com/geirsson/codegen/Codegen.scala (hunk 1 above)
Comment (nightscape) [reply src/main/scala/com/geirsson/codegen/Codegen.scala]: Good note! I'll change that.

**Comment by nightscape** [comment]
The idea of generating type aliases sounds good, too.
If e.g. you have a custom data type called `circle` then it makes sense to have an explicit type alias `Circle` for that instead of just a `(Double, Double)`. I would go for CamelCase in order to generate code that sticks to Scala naming conventions.
I think it's still a good idea to be able to specify a type-map initially and have a good set of defaults.
Type aliases like

```
type Int64 = Int
type VarChar = String
```

would just create a leaky abstraction.

But maybe you could merge this PR independently of the above ideas (I amended the test change and the use of the `warn` method to the commit), just so you have a way of quickly testing scala-db-codegen without having to run it n times for n unknown column types.

**Comment by olafurpg** [comment]
LGTM 👍 

> I think it's still a good idea to be able to specify a type-map initially and have a good set of defaults.

Totally agree. The current defaults are only the ones required for my personal project. It should be possible to provide defaults for most of the Postgres data built-in types: https://www.postgresql.org/docs/9.5/static/datatype.html#DATATYPE-TABLE

> Type aliases like [...] would just create a leaky abstraction.

They don't necessarily have to be type aliases. You could in theory have any class with custom decoders/encoders.

**Pull request merged**